src/gui/layout/qgslayoutchartwidget.cpp

```diff
@@ -0,0 +1,398 @@
+/***************************************************************************
+                         qgslayoutchartwidget.cpp
+                         --------------------------
+     begin                : August 2025
+     copyright            : (C) 2025 by Mathieu
+     email                : mathieu at opengis dot ch
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgslayoutchartwidget.h"
+#include "moc_qgslayoutchartwidget.cpp"
+#include "qgsapplication.h"
+#include "qgsgui.h"
+#include "qgslayout.h"
+#include "qgslayoutchartseriesdetailswidget.h"
+#include "qgslayoutitemchart.h"
+#include "qgsplotregistry.h"
+#include "qgsplotwidget.h"
+
+
+QgsLayoutChartWidget::QgsLayoutChartWidget( QgsLayoutItemChart *chartItem )
+  : QgsLayoutItemBaseWidget( nullptr, chartItem )
+  , mChartItem( chartItem )
+{
+  setupUi( this );
+
+  //add widget for general composer item properties
+  mItemPropertiesWidget = new QgsLayoutItemPropertiesWidget( this, chartItem );
+  mainLayout->addWidget( mItemPropertiesWidget );
+
+  QMap<QString, QString> plotTypes = QgsApplication::instance()->plotRegistry()->plotTypes();
+  for ( auto plotTypesIterator = plotTypes.keyValueBegin(); plotTypesIterator != plotTypes.keyValueEnd(); ++plotTypesIterator )
+  {
+    mChartTypeComboBox->addItem( plotTypesIterator->second, plotTypesIterator->first );
+  }
+
+  mLayerComboBox->setFilters( Qgis::LayerFilter::VectorLayer );
+
+  connect( mChartTypeComboBox, static_cast<void ( QComboBox::* )( int )>( &QComboBox::currentIndexChanged ), this, &QgsLayoutChartWidget::mChartTypeComboBox_currentIndexChanged );
+  connect( mChartPropertiesButton, &QPushButton::clicked, this, &QgsLayoutChartWidget::mChartPropertiesButton_clicked );
+
+  connect( mLayerComboBox, &QgsMapLayerComboBox::layerChanged, this, &QgsLayoutChartWidget::changeLayer );
+  connect( mLayerComboBox, &QgsMapLayerComboBox::layerChanged, mSortExpressionWidget, &QgsFieldExpressionWidget::setLayer );
+  connect( mSortCheckBox, &QCheckBox::stateChanged, this, &QgsLayoutChartWidget::mSortCheckBox_stateChanged );
+  connect( mSortExpressionWidget, static_cast<void ( QgsFieldExpressionWidget::* )( const QString &, bool )>( &QgsFieldExpressionWidget::fieldChanged ), this, &QgsLayoutChartWidget::changeSortExpression );
```

Comment:
should have expression context generator registered

Reply:
Yeah, planning to do that alongside data-defined properties in a follow up PR, this one is big enough as it is :)